### PR TITLE
Improving `asType`

### PR DESCRIPTION
**Pull request by tomaszpolanski**

### CODE REVIEW

File: kotlin-options/src/main/kotlin/com/petertackage/kotlinoptions/Option.kt [hunk 1]
```diff
@@ -82,9 +82,9 @@ sealed class Option<out T : Any> {
         }
     }
 
-    fun <R : Any> asType(clazz: Class<R>): Option<R> {
+    inline fun <reified R : Any> asType(): Option<R> {
         return when (this) {
-            is Some -> if (clazz.isInstance(value)) Some(clazz.cast(value)) else None
+            is Some -> if (value is R) optionOf(value as R) else None
             is None -> this
```
Comment (peter-tackage) [inline kotlin-options/src/main/kotlin/com/petertackage/kotlinoptions/Option.kt:88]: Can you please add a test for the `assertThat(None.asType()).isEqualTo(None)` case.

**Review by peter-tackage — CHANGES_REQUESTED** [review]
Conditional merge.

**Review by peter-tackage — APPROVED** [review]

**Comment by peter-tackage** [comment]
Thanks!
